bokeh/io.py

```diff
@@ -439,17 +434,13 @@ def _get_save_args(state, filename, resources, title):
 
     return filename, resources, title
 
-def _save_helper(obj, filename, resources, title, validate):
+def _save_helper(obj, state, filename, resources, title, validate):
     remove_after = False
-    if isinstance(obj, LayoutDOM):
-        if obj.document is None:
-            Document().add_root(obj)
-            remove_after = True
-        doc = obj.document
-    elif isinstance(obj, Document):
-        doc = obj
-    else:
-        raise RuntimeError("Unable to save object of type '%s'" % type(obj))
+    doc = state.document
+
+    if obj.document is None:
+        doc.add_root(obj)
+        remove_after = True
 
     if validate:
         doc.validate()
```

Comment:
@canavandl just checking, decisions was to remove all the above code and just call `standalone_html_page_for_models` with `obj`, directly at the start of `_save_helpers`, since it uses `_ModelInDocument` internally?